### ISSUE TITLE
Building already built project results in 1 succeeded instead of 1 up-to-date

### ISSUE THREAD

**Issue by jakubsuchybio** [Fody/Costura#326]
#### Describe the bug
Given I have 1 project in solution **without** Costura.Fody installed
When I build project for the first time
Then It results in 1 succeeded project built
And When I build the same project for the second time
Then it results in 1 up-to-date project built

Given I have 1 project in solution **with** Costura.Fody installed
When I build project for the first time
Then It results in 1 succeeded project built
And When I build the same project for the second time
Then it **results in 1 succeeded instead of 1 up-to-date** project built

#### Minimal Repro
see Description

#### Submit a PR that fixes the bug
I have no idea where to start to fix this issue. I could try it, with some navigation from you.

#### Expected behavior
I would expect that consecutive builds without any changes to the project would result in 1 up-to-date build. 

#### Additional context
When I set Output verbosity to 'Normal' in VS settings I get this part of output
```
1>GenerateTargetFrameworkMonikerAttribute:
1>Skipping target "GenerateTargetFrameworkMonikerAttribute" because all output files are up-to-date with respect to the input files.
1>CoreCompile:
1>Skipping target "CoreCompile" because all output files are up-to-date with respect to the input files.
1>FodyTarget:
1>Skipping target "FodyTarget" because all output files are up-to-date with respect to the input files.
1>_CopyAppConfigFile:
1>Skipping target "_CopyAppConfigFile" because all output files are up-to-date with respect to the input files.
1>CopyFilesToOutputDirectory:
1>  CP.LogAdapter -> c:\DokumentySVN\MEWProjects\MEW\CP.LogAdapter\bin\Debug\CP.LogAdapter.dll
1>
```
Here you can see that FodyTarget is skipped, because target is up-to-date, however step with Copying the resulting assembly is still made. By my understanding this is not necessary and this step should be skipped as well.

**Comment by SimonCropp** [comment]
yep sounds like this should be fixed. please submit a PR that fixes it

**Comment by jakubsuchybio** [comment]
@SimonCropp Sorry, tried to figure it out, narrowed it down to Fody Nuget into Fody.targets, where is FodyTarget and FodyUpdateCopyLocalFilesTarget which I think is that procedure that copies the final DLL into OutputDir. But I don't know how I would change it's implementation in UpdateReferenceCopyLocalTask to not copy.

**Comment by SimonCropp** [comment]
In that case before we start, has you or your company contributed to the fody collective 
https://github.com/Fody/Fody#preamble

**Comment by jakubsuchybio** [comment]
I now did :)
https://opencollective.com/jakub-suchy

**Comment by SimonCropp** [comment]
@jakubsuchybio can u upload a solution that reproduces the issue https://github.com/Fody/Costura/blob/master/.github/ISSUE_TEMPLATE/bug_report.md#minimal-repro

**Comment by jakubsuchybio** [comment]
https://github.com/jakubsuchybio/fody-costura-build-repro
Here it is.
I should mention that when I was creating this repro, when I used PackageReference it worked fine.
Only when using packages.config it results in "always building" and not getting it "up-to-date"

**Comment by SimonCropp** [comment]
Does it only happen woth costura? Or does it also happen with other fody weavers?

**Comment by tom-englert** [comment]
@jakubsuchybio Also is this related to PackageReference VS packages.config, or maybe rather to old VS new project format?

**Comment by jakubsuchybio** [comment]
@SimonCropp Seems like a problem in Fody.
I tried to uninstall Costura and installed EmptyConstructor weaver and it still builds everytime instead of saying it is up-to-date.

@tom-englert I don't know. All I can say for sure is that new PackageReference does not have this bug. Only old format packages.config has it.

**Comment by tom-englert** [comment]
Since Fody does exactly the same in both scenarios, I would rather think this problem is related to MSBuild

**Comment by jakubsuchybio** [comment]
If that is so, where should I move this issue ticket to? I don't think that MSFT has some repo for msbuild issues for old packages.config format. Well I guess it is no issue then. Packages.config is obsolete anyway. It is just a cosmetic issue I'd say.
